src/emailcontactproxymodel.cpp

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2020- 2021 Open Mobile Platform LLC.
+ *
+ * This program is licensed under the terms and conditions of the
+ * Apache License, version 2.0.  The full text of the Apache License is at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+#include "emailcontactproxymodel.h"
+#include "emailcontactmodel.h"
+
+
+EmailContactProxyModel::EmailContactProxyModel(QObject *parent)
+    : QSortFilterProxyModel(parent)
+{
+    EmailContactModel *model = new EmailContactModel(this);
+    setSourceModel(model);
+    QSortFilterProxyModel::setSortRole(EmailContactModel::TimeStampRole);
+    QSortFilterProxyModel::sort(0, Qt::SortOrder::AscendingOrder);
+
+    m_regexp.setPatternSyntax(QRegExp::Wildcard);
+    m_regexp.setCaseSensitivity(Qt::CaseInsensitive);
+    m_regexp.setPattern("*");
+}
+
+
+QHash<int, QByteArray> EmailContactProxyModel::roleNames() const
+{
+    if (QAbstractItemModel *source = sourceModel())
+        return source->roleNames();
+    return QHash<int, QByteArray>();
+}
+
+
+bool EmailContactProxyModel::filterAcceptsRow(int source_row, const QModelIndex &source_parent) const
+{
+    if (this->sourceModel() == nullptr)
+        return false;
+
+    QModelIndex index = sourceModel()->index(source_row, 0, source_parent);
+    if (!index.isValid())
+        return false;
+
+    const QString text = sourceModel()->data(index, EmailContactModel::NameRole).toString();
+    return m_regexp.exactMatch(text);
+}
+
+
+int EmailContactProxyModel::count() const
+{
+    return rowCount();
+}
+
+
+QString EmailContactProxyModel::filter() const
+{
+    return m_filter;
+}
+
+
+void EmailContactProxyModel::setMaxCount(const int maxCount)
+{
+    EmailContactModel *model = qobject_cast<EmailContactModel*>(sourceModel());
+    Q_ASSERT(model);
+    model->setMaxCount(maxCount);
+}
+
+
+void EmailContactProxyModel::setFilter(const QString &filter)
+{
+    if (m_filter != filter) {
+        m_filter = filter;
+        invalidateFilter();
```

Comment:
Shouldn't this be called after the real filter, the regexp, has been updated?

Reply:
Yes you are right.
done